### PR TITLE
Opaque pointers: Part of changes in postProcessRowMajorMatrix

### PR DESCRIPTION
**Pull request by mariusz-sikora-at-amd**

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 10f56966b9ec8dde4a15063a48226ca101452ebb
<details><summary>CTS tests (Failed: 0/216059)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 35298/73410 (48.1%)</li>
<li>Failed: 0/73410 (0.0%)</li>
<li>Not Supported: 38112/73410 (51.9%)</li><li>Warnings: 0/73410 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 33017/69239 (47.7%)</li>
<li>Failed: 0/69239 (0.0%)</li>
<li>Not Supported: 36222/69239 (52.3%)</li><li>Warnings: 0/69239 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37553/73410 (51.2%)</li>
<li>Failed: 0/73410 (0.0%)</li>
<li>Not Supported: 35857/73410 (48.8%)</li><li>Warnings: 0/73410 (0.0%)</li>
</ul></details>


</ul></details>

File: llpc/translator/lib/SPIRV/SPIRVReader.cpp [hunk 1]
```diff
@@ -1287,7 +1289,9 @@ bool SPIRVToLLVM::postProcessRowMajorMatrix() {
 
             for (unsigned i = 0; i < cast<FixedVectorType>(pointerType)->getNumElements(); i++) {
               Value *const pointerElem = getBuilder()->CreateExtractElement(pointer, i);
-              Type *const newLoadElemType = pointerElem->getType()->getPointerElementType();
+              Type *const newLoadElemType = pointerElemType;
+              // TODO: Remove this when LLPC will switch fully to opaque pointers.
+              assert(IS_OPAQUE_OR_POINTEE_TYPE_MATCHES(pointerElem->getType(), newLoadElemType));
```
Comment (nhaehnle) [inline llpc/translator/lib/SPIRV/SPIRVReader.cpp:1294]: newLoadElemType is always equal to pointerElemType, right? If so, can you just remove that variable?

File: llpc/translator/lib/SPIRV/SPIRVReader.cpp [hunk 2]
```diff
@@ -1334,7 +1340,9 @@ bool SPIRVToLLVM::postProcessRowMajorMatrix() {
             load->replaceAllUsesWith(getBuilder()->CreateTransposeMatrix(newLoad));
           } else {
             // Otherwise we are loading a single element and it's a simple load.
-            Type *const newLoadType = pointer->getType()->getPointerElementType();
+            Type *const newLoadType = pointerElemType;
+            // TODO: Remove this when LLPC will switch fully to opaque pointers.
+            assert(IS_OPAQUE_OR_POINTEE_TYPE_MATCHES(pointer->getType(), newLoadType));
```
Comment (nhaehnle) [inline llpc/translator/lib/SPIRV/SPIRVReader.cpp:1345]: Also here, can newLoadType be removed?

File: llpc/translator/lib/SPIRV/SPIRVReader.cpp (hunk 1 above)
Comment (mariusz-sikora-at-amd) [reply llpc/translator/lib/SPIRV/SPIRVReader.cpp]: Thanks ! pushed update.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 152666d99f59fb807b5c58bc03ee9b7dc8296023
<details><summary>CTS tests (Failed: 0/216059)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 35298/73410 (48.1%)</li>
<li>Failed: 0/73410 (0.0%)</li>
<li>Not Supported: 38112/73410 (51.9%)</li><li>Warnings: 0/73410 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 33017/69239 (47.7%)</li>
<li>Failed: 0/69239 (0.0%)</li>
<li>Not Supported: 36222/69239 (52.3%)</li><li>Warnings: 0/69239 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37553/73410 (51.2%)</li>
<li>Failed: 0/73410 (0.0%)</li>
<li>Not Supported: 35857/73410 (48.8%)</li><li>Warnings: 0/73410 (0.0%)</li>
</ul></details>


</ul></details>

**Review by nhaehnle — APPROVED** [review]
LGTM

**Pull request merged**